[EDU-2697] feat: add status section to edge caching (#534)
aziontech/docs: src/content/docs/en/pages/main-menu/reference/build/edge-application/edge-caching.mdx

---
title: Edge Caching
description: Edge Caching speeds up content delivery by keeping your content cached at the edge and bringing your application closer to your users.
meta_tags: caching, edge, edge caching, edge computing, performance, cache status, bypass, hit, miss
namespace: documentation_products_edge_caching
permalink: /documentation/products/edge-application/edge-caching/
---
import Button from '~/components/Button.astro'

> **Edge Caching** is a module available for **Edge Application** that allows you to configure how your application caches content, reducing latency and increasing transfer rates through Azion's network of edge nodes. This not only enhances performance and scalability for your content, but also cuts back on infrastructure costs.

Reviewer: ```suggestion
**Edge Caching** is a module available for **Edge Application** that allows you to configure how your application caches content, reducing latency and increasing transfer rates through Azion's network of edge nodes. This not only enhances performance and scalability for your content but also cuts back on infrastructure costs.
```

> With Azion, you can customize the time the cache remains in the edge through *time-to-live* (TTL) values. Cache policies with large TTL values can optimize the performance of your application, improving the experience for your users, reducing traffic with Azion, and reducing the load at your origin. However, a low cache TTL can help you ensure end-users always see the most current information.

Reviewer 2: ```suggestion
With Azion, you can customize the time the cache remains in the edge through *time-to-live* (TTL) values. Cache policies with large TTL values can optimize the performance of your application, improving your users experience, reducing traffic with Azion, and reducing the load at your origin. However, a low cache TTL can help you ensure end-users always see the most current information.
```

---

## Cache status

Azion provides a layer to cache content in the edge network, which stands between origin servers and the end user. When content is successfully cached at the edge, it receives a HIT status and it can be directly delivered to your users from the edge node nearest to them, eliminating the frequency of access to origin servers.

> In the event of a MISS status, when the requested content isn't available in cache, the Edge Caching module minimizes the impact by maintaining a keep-alive connection with the origin server whenever possible. This approach avoids the overhead of frequent TCP/IP handshakes. Regardless of the number of simultaneous requests made to Azion's edge nodes, each edge node only needs to request content from the origin once per MISS.

Author: ```suggestion
In the event the requested content isn't available in cache, the Edge Caching module minimizes the impact by maintaining a keep-alive connection with the origin server whenever possible. This approach avoids the overhead of frequent TCP/IP handshakes. Regardless of the number of simultaneous requests made to Azion's edge nodes, each edge node only needs to request content from the origin once per MISS or other statuses.
```

```bash
X-Cache: HIT from 200.000.000.00 with HTTP/2.0
```

:::note
Find out more about how to obtain other cache data in the [How to verify application cache indicators using Modheader for Google Chrome](/en/documentation/products/guides/check-page-cache-time/) guide.
:::

### Status definitions

| Status | Definition |
| --- | --- |
| **HIT** | Valid and up-to-date content is supplied directly from the edge cache. |
| **MISS** | If the requested content isn't found in the cache at the edge, it's fetched from the origin. Through the response, the content might be cached for future requests. |
| **EXPIRED** | The cached content has expired the TTL set at the edge. When the origin responds, the cached content is updated for future requests. |
| **STALE** | When choosing to serve [stale cache](#stale-cache), if the origin fails to respond and the edge cache has expired, a stale version of the content is served. |
| **UPDATING** | The cached content has expired and stale cache is being served, but the content will be updated for future requests. |

> | **BYPASS** | The edge requests the content from the origin directly instead of using a cached version due to the active [Bypass Cache](/en/documentation/products/edge-application/rules-engine/#bypass-cache) behavior. |

Author: ```suggestion
| **BYPASS** | The edge requests the content from the origin directly instead of using a cached version due to the active [Bypass Cache](/en/documentation/products/edge-application/rules-engine/#bypass-cache) behavior. |
| **-** | If no status is received, the content requested is restricted from caching. For instance, if the requested content is a `POST` request and [Caching for POST](/en/documentation/products/edge-application/cache-settings/#caching-http-methods) is disabled, the status is not logged. |
```

---

## Cache status

Azion provides a layer to cache content in the edge network, which stands between origin servers and the end user. When content is successfully cached at the edge, it receives a [cache status HIT](#status-definitions) and it can be directly delivered to your users from the edge node nearest to them, reducing the frequency of access to origin servers.

> In the event the requested content isn't available in cache, the Edge Caching module minimizes the impact by maintaining a keep-alive connection with the origin server whenever possible. This approach avoids the overhead of frequent TCP/IP handshakes. Regardless of the number of simultaneous requests made to Azion's edge nodes, each edge node only needs to request content from the origin once per MISS or other statuses.

Reviewer 3: Não é minha vertical aqui (Build), mas que eu saiba o módulo de Edge Caching não tem nada a ver com "mantaining a keep-alive connection with the origin".

Caching é capacidade de entregar respostas direto da edge, sem precisar sequer conectar na origem. Quando o cache precisa ser ATUALIZADO, aí sim uma conexão é feita à origem, mas o keep-alive não tem nenhuma relação direta com a configuração de Edge Caching em si...

Talvez, querem citar aqui o controle de `Thundering Herd`, que na verdade é um controle que permite que múltiplas requests ao mesmo tempo para um mesmo objeto gerem apenas 1 conexão para a origem. Tem que validar com o PM esse parágrafo.

Author: @Vitor-Azion

Reviewer 4: Eu adicionaria no texto as duas práticas:

"When the requested content is not available in the cache, Azion's Edge Caching module employs strategies to minimize the impact on the origin server and optimize content delivery. One of these strategies includes maintaining keep-alive connections with the origin server whenever possible, to reduce the overhead associated with frequent TCP/IP handshakes. This is particularly effective during cache updates or in situations of a cache MISS. Additionally, we implement Thundering Herd control to ensure that, in the face of multiple simultaneous requests for the same non-cached object, only one connection is established with the origin server. This approach ensures that each of Azion's edge nodes requests content from the origin server only once per MISS or other non-HIT statuses, thereby optimizing network efficiency and reducing the load on the origin server."

Reviewer 4: @bru-andrade

> In the event the requested content isn't available in cache, the Edge Caching module minimizes the impact by maintaining a keep-alive connection with the origin server whenever possible. This approach avoids the overhead of frequent TCP/IP handshakes. Regardless of the number of simultaneous requests made to Azion's edge nodes, each edge node only needs to request content from the origin once per MISS or other statuses.

Author: ```suggestion
When the requested content is not available in the cache, Azion's Edge Caching module employs strategies to minimize the impact on the origin server and optimize content delivery. One of these strategies includes maintaining keep-alive connections with the origin server whenever possible, to reduce the overhead associated with frequent TCP/IP handshakes. This is particularly effective during cache updates or in situations of a [cache MISS](#status-definitions). Additionally, we implement Thundering Herd control to ensure that, in the face of multiple simultaneous requests for the same non-cached object, only one connection is established with the origin server. This approach ensures that each of Azion's edge nodes request content from the origin server only once per MISS or other non-HIT statuses, thereby optimizing network efficiency and reducing the load on the origin server.
```

> When the requested content is not available in the cache, Azion's Edge Caching module employs strategies to minimize the impact on the origin server and optimize content delivery. One of these strategies includes maintaining keep-alive connections with the origin server whenever possible, to reduce the overhead associated with frequent TCP/IP handshakes. This is particularly effective during cache updates or in situations of a [cache MISS](#status-definitions). Additionally, we implement Thundering Herd control to ensure that, in the face of multiple simultaneous requests for the same non-cached object, only one connection is established with the origin server. This approach ensures that each of Azion's edge nodes request content from the origin server only once per MISS or other non-HIT statuses, thereby optimizing network efficiency and reducing the load on the origin server.

Author: @arturdj parágrafo providenciado pelo incrível @Vitor-Azion!

You can obtain the cache status of an edge application's resource by requesting the content URI and adding the Azion debug header `Pragma: azion-debug-cache`. A successful response will return an `X-Cache` header with the cache status, the IP of the edge node, and the protocol used in the request:

```bash
X-Cache: HIT from 200.000.000.00 with HTTP/2.0
```

:::note
Find out more about how to automate your browser to obtain other cache data in the [How to verify application cache indicators using Modheader for Google Chrome](/en/documentation/products/guides/check-page-cache-time/) guide.
:::

### Status definitions

| Status | Definition |
| --- | --- |

> | **HIT** | Valid and up-to-date content is supplied directly from the edge cache. |
> | **MISS** | If the requested content isn't found in the cache at the edge, it's fetched from the origin. Through the response, the content might be cached for future requests. |
> | **EXPIRED** | The cached content has expired the TTL set at the edge. When the origin responds, the cached content is updated for future requests. |
> | **STALE** | When choosing to serve [stale cache](#stale-cache), if the origin fails to respond and the edge cache has expired, a stale version of the content is served. |
> | **UPDATING** | The cached content has expired and stale cache is being served, but the content is being updated in the origin. |
> | **BYPASS** | The edge requests the content from the origin directly instead of using a cached version due to the active [Bypass Cache](/en/documentation/products/edge-application/rules-engine/#bypass-cache) behavior. |

Reviewer: ```suggestion
| **HIT** | Valid and up-to-date content is supplied directly from the edge cache |
| **MISS** | If the requested content isn't found in the cache at the edge, it's fetched from the origin. Through the response, the content might be cached for future requests |
| **EXPIRED** | The cached content has expired the TTL set at the edge. When the origin responds, the cached content is updated for future requests |
| **STALE** | When choosing to serve [stale cache](#stale-cache), if the origin fails to respond and the edge cache has expired, a stale version of the content is served |
| **UPDATING** | The cached content has expired and a stale cache is being served, but the content is being updated in the origin |
| **BYPASS** | The edge requests the content from the origin directly instead of using a cached version due to the active [Bypass Cache](/en/documentation/products/edge-application/rules-engine/#bypass-cache) behavior |
```

## Stale cache

When your application's cache expires, it becomes *stale* (obsolete) and the server searches the origin for the most up-to-date version. This new version, with an updated expiration date, overwrites the old one.

> However, the server may encounter problems and be unable to get the most up-to-date version of your application from the origin. When active, **Stale Cache** delivers the latest version of the content in cache, regardless of expiration, whenever origin servers are unavailable.

Reviewer 3: Reading again now, maybe a simple change does the job:

from: `When active, **Stale Cache** delivers the latest version of the content in cache...`
to: `When active, **Stale Cache** delivers the latest **available** version of the content in cache...`

> However, the server may encounter problems and be unable to get the most up-to-date version of your application from the origin. When active, **Stale Cache** delivers the latest version of the content in cache, regardless of expiration, whenever origin servers are unavailable.

Author: ```suggestion
However, the server may encounter problems and be unable to get the most up-to-date version of your application from the origin. When active, **Stale Cache** delivers the last available version of the content in cache, regardless of expiration, whenever origin servers are unavailable.
```

## Stale cache

**Stale cache** is an optimization to guarantee content delivery when a cache object is not used frequently. Traditionally, if the cached content expires or cannot be located, the edge requests content directly from origin servers.

> When stale cache is active and origin servers cannot be reached, a version of the content with expired TTL and the STALE status can be delivered for the request. The next successful request to the origin for the updated content then receives a HIT status.

Author: @arturdj Ajustada a descrição

Reviewer 3: Not very accurate. Please re-check with Product Manager

## Stale cache

**Stale cache** is an optimization to guarantee content delivery when a cache object is not used frequently. Traditionally, if the cached content expires or cannot be located, the edge requests content directly from origin servers.

When stale cache is active and origin servers cannot be reached, a version of the content with expired TTL and the STALE status can be delivered for the request. The next successful request to the origin for the updated content then receives a HIT status.

> However, the server may encounter problems and be unable to get the most up-to-date version of your application from the origin. When active, **Stale Cache** delivers the last available version of the content in cache, regardless of expiration, whenever origin servers are unavailable.

Author: ```suggestion
```
